src/lib/libhaproxy.py

```diff
@@ -150,7 +150,10 @@ def process_configs(self, configs):
             # nicer HTTP checks for HTTP backends will also
             # be enabled to perform HTTP requests as part of
             # checking backend health
-            attributes = ['check fall 3 rise 2']
+            if config['check']:
+                attributes = ['check fall 3 rise 2']
+            else:
+                attributes = []
```

Comment:
What about just setting attributes = [], and then if config['check']: is true, setting the check settings?